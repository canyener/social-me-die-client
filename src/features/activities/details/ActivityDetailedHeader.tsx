import React from 'react'
import { observer } from 'mobx-react-lite';
import { Link } from 'react-router-dom';
import { Segment, Image, Item, Header, Button } from 'semantic-ui-react'
import { format } from 'date-fns'

import { IActivity } from '../../../app/models/activity';

const activityImageStyle = {
  filter: 'brightness(30%)'
};

const activityImageTextStyle = {
  position: 'absolute',
  bottom: '5%',
  left: '5%',
  width: '100%',
  height: 'auto',
  color: 'white'
};

const ActivityDetailedHeader: React.FC<{activity: IActivity}> = ({ activity }) => {
  return (
    <Segment.Group>
    <Segment basic attached='top' style={{ padding: '0' }}>
      <Image
        src={`/assets/categoryImages/${activity.category}.jpg`}
        fluid
        style={activityImageStyle}
      />
      <Segment style={activityImageTextStyle} basic>
        <Item.Group>
          <Item>
            <Item.Content>
              <Header
                size='huge'
                content={activity.title}
                style={{ color: 'white' }}
              />
              <p>{format(activity.date,'eeee do MMMM')}</p>
              <p>
                Hosted by <strong>Cancan</strong>
              </p>
            </Item.Content>
          </Item>
        </Item.Group>
      </Segment>
    </Segment>
    <Segment clearing attached='bottom'>
      <Button color='teal'>Join Activity</Button>
      <Button>Cancel attendance</Button>
      <Button as={Link} to={`/manage/${activity.id}`} color='orange' floated='right'>
        Manage Event
      </Button>
    </Segment>
  </Segment.Group>
  )
}

export default observer(ActivityDetailedHeader)