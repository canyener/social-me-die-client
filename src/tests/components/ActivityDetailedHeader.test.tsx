import { shallow } from 'enzyme'
import ActivityDetailedHeader from '../../features/activities/details/ActivityDetailedHeader'
import { fakeActivities } from '../fixtures/seed'

test('Should render ActivityDetailedHeader correctly', () => {
  const wrapper = shallow(<ActivityDetailedHeader activity={fakeActivities[0]} />)

  expect(wrapper).toMatchSnapshot()
})