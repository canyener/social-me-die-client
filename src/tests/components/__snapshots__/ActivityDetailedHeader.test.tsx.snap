// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ActivityDetailedHeader correctly 1`] = `
<SegmentGroup>
  <Segment
    attached="top"
    basic={true}
    style={
      Object {
        "padding": "0",
      }
    }
  >
    <Image
      as="img"
      fluid={true}
      src="/assets/categoryImages/Fake Category 1.jpg"
      style={
        Object {
          "filter": "brightness(30%)",
        }
      }
      ui={true}
    />
    <Segment
      basic={true}
      style={
        Object {
          "bottom": "5%",
          "color": "white",
          "height": "auto",
          "left": "5%",
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <ItemGroup>
        <Item>
          <ItemContent>
            <Header
              content="Fake Activity 1"
              size="huge"
              style={
                Object {
                  "color": "white",
                }
              }
            />
            <p>
              Friday 24th December
            </p>
            <p>
              Hosted by 
              <strong>
                Cancan
              </strong>
            </p>
          </ItemContent>
        </Item>
      </ItemGroup>
    </Segment>
  </Segment>
  <Segment
    attached="bottom"
    clearing={true}
  >
    <Button
      as="button"
      color="teal"
    >
      Join Activity
    </Button>
    <Button
      as="button"
    >
      Cancel attendance
    </Button>
    <Button
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      color="orange"
      floated="right"
      to="/manage/1"
    >
      Manage Event
    </Button>
  </Segment>
</SegmentGroup>
`;
